app/controllers/posts_controller.rb

```diff
@@ -1,5 +1,6 @@
 class PostsController < ApplicationController
   before_filter :load_post
+  before_filter :load_comments
```

Comment:
No. I removed this filter and associated private method, and defined @comments in the show method.